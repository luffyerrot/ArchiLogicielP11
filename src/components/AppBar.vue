<template>
    <v-app-bar>
        <template v-slot:prepend>
            <v-app-bar-nav-icon
                variant="text"
                @click.stop="drawer = !drawer"
            />
        </template>
        <template v-slot:append>
            <v-btn
                :prepend-icon="props.theme === 'light' ? 'mdi-weather-sunny' : 'mdi-weather-night'"
                text="Changer le thÃ¨me"
                slim
                @click="$emit('changeTheme')"
            />
            <p class="mx-4">{{ user.mail }}</p>
        </template>
        <v-app-bar-title
            class="clickable"
            text="Service MedHead"
            @click="goTo('/')"
        />
    </v-app-bar>
    <v-navigation-drawer
        v-model="drawer"
        temporary
    >
        <v-list
            v-for="(item, index) in items"
            :key="'menu' + index"
        >
            <v-list-item
                :title="item.title"
                @click="goTo(item.destination)"
            />
        </v-list>
    </v-navigation-drawer>
</template>

<script setup lang="ts">
    import router from '@/router';
    import { useUserStore } from '@/stores/user';

    defineEmits(['changeTheme'])

    const props = defineProps<{
        theme: string
    }>()

    const user = useUserStore();

    const drawer = ref(false);
    const items = computed(() => {
        const baseItems = [
            {
                title: 'Accueil',
                destination: '/'
            },
            {
                title: 'Recherche',
                destination: '/mapfinder'
            }
        ];
        const loggedInItems = [
            {
                title: 'S\'enregistrer',
                destination: '/register'
            },
            {
            title: 'Connexion',
            destination: '/login'
            }
        ];
        const loggedOutItems = [
            {
                title: 'Se deconnecter',
                destination: '/logout'
            }
        ];
        return [...(user.isLogged() ? loggedOutItems : loggedInItems), ...baseItems]
    });

    function goTo(destination: string) {
        router.push(destination);
    }
</script>

<style scoped lang="scss">
    .clickable {
        cursor: pointer;
    }
</style>