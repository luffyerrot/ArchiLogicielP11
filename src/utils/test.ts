import { ComponentMountingOptions, mount, shallowMount, VueWrapper } from "@vue/test-utils";
import { createPinia, setActivePinia } from "pinia";
import { expect, it } from "vitest";
import { createVuetify } from "vuetify";
import * as components from 'vuetify/components'
import * as directives from 'vuetify/directives'

export function factory(originalComponent: any, shallow = false, pinia = false, options?: ComponentMountingOptions<any, unknown>): VueWrapper<unknown, any> {
    if (pinia) {
        setActivePinia(createPinia());
    }
    if (shallow) {
        return shallowMount(originalComponent, options);
    } else {
        const vuetify = createVuetify({
            components,
            directives,
        });
        const defaultOptions = {
            global: {
                plugins: [vuetify]
            },
            ...options
        };
        return mount(originalComponent, defaultOptions);
    }
};

export function renderCorrectly(wrapper: VueWrapper<unknown, any>) {
    it('Bon affichage de la page', () => {
        expect(wrapper.html()).toMatchSnapshot();
    })
}