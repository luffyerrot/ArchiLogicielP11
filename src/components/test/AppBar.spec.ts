import { factory, renderCorrectly } from "@/utils/test";
import { describe } from "vitest";
import AppBar from "../AppBar.vue";

describe('Barre du header', () => {    
    const wrapper = factory(AppBar, true, true);
    renderCorrectly(wrapper);
})