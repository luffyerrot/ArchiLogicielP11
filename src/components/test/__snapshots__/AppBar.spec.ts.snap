// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Barre du header > Bon affichage de la page 1`] = `
"<v-app-bar-stub data-v-d1b168af="" modelvalue="true" location="top" absolute="false" collapse="false" density="default" extended="false" extensionheight="48" flat="false" floating="false" height="64" border="false" tile="false" tag="header" order="0" scrollthreshold="300"></v-app-bar-stub>
<v-navigation-drawer-stub data-v-d1b168af="" disableresizewatcher="false" disableroutewatcher="false" expandonhover="false" floating="false" modelvalue="false" permanent="false" railwidth="56" scrim="true" temporary="true" persistent="false" touchless="false" width="256" location="start" sticky="false" border="false" order="0" absolute="false" tile="false" tag="nav"></v-navigation-drawer-stub>"
`;
